keyboards/ah/haven65/keymaps/via/keymap.c

```diff
@@ -0,0 +1,206 @@
+      KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,
+      KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS
+  ),
+  [3] = LAYOUT(
+      KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
+      KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,
+      KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS
+  )
+};
+
+//indicator variables
+bool caps_is_active = false;
+int mode = 5;
+
+//to change color if the caps lock is on
+void freather_set_color(bool capslock_is_on);
+bool process_record_user(uint16_t keycode, keyrecord_t *record);
+
+
+
+// Declaring a _freather_config_t struct that will store data
+typedef struct _indicator_config_t {
+    uint8_t h;
+    uint8_t s;
+    uint8_t v;
+    bool enabled;
+} indicator_config;
+
+// Board memory layout
+typedef struct _freather_config_t {
+    indicator_config caps;
+} freather_config;
+
+
+
+// Check if the size of the reserved persistent memory is the same as the size of struct freather_config
+_Static_assert(sizeof(indicator_config) == EECONFIG_USER_DATA_SIZE, "Mismatch in keyboard EECONFIG stored data");
+
+// Declaring a new variable freather of type freather_config
+freather_config freather;
+
+// Declaring enums for VIA config menu
+enum via_freather_enums {
+
+    id_freather_indicator_enabled = 1,
+    id_freather_indicator_color = 2
+
+};
+
+// Initializing persistent memory configuration: default values are declared and stored in PMEM
+void eeconfig_init_user(void) {
+    // Default values
+    freather.caps.h       = 100;
+    freather.caps.s       = 100;
+    freather.caps.v       = 100;
+    freather.caps.enabled = true;
+    rgblight_set_effect_range(0,1);
+    // Write default value to EEPROM now
+    eeconfig_update_user_datablock(&freather);
+}
+
+// On Keyboard startup
+void keyboard_post_init_user(void) {
+    // Read custom menu variables from memory
+    eeconfig_read_user_datablock(&freather);
+}
+
+// Handle the data received by the keyboard from the VIA menus
+void freather_config_set_value(uint8_t *data) {
```

Comment:
All of this via stuff is repetitive, and could/should be moved to a common file.

Reply:
> All of this via stuff is repetitive, and could/should be moved to a common file.

so like a .c file in the main keyboard folder?